hamilton/execution/graph_functions.py

```diff
@@ -200,41 +232,29 @@ def dfs_traverse(
             except Exception as e:
                 success = False
                 error = e
-                # This code is coupled to how @config resolution works. Ideally it shouldn't be,
-                # so when @config resolvers are changed to return Nodes, then fn.__name__ should
-                # just work.
-                original_func_name = "unknown"
-                if node_.originating_functions:
-                    if hasattr(node_.originating_functions[0], "__original_name__"):
-                        original_func_name = node_.originating_functions[0].__original_name__
-                    else:
-                        original_func_name = node_.originating_functions[0].__name__
-                module = (
-                    node_.originating_functions[0].__module__
-                    if node_.originating_functions
-                    and hasattr(node_.originating_functions[0], "__module__")
-                    else "unknown_module"
-                )
-                message = f"> {node_.name} [{module}.{original_func_name}()] encountered an error"
-                padding = " " * (80 - len(message) - 1)
-                message += padding + "<"
-                input_string = create_input_string(kwargs)
-                message += "\n> Node inputs:\n" + input_string
-                border = "*" * 80
-                logger.exception("\n" + border + "\n" + message + "\n" + border)
+                step = "[pre-node-execute]" if pre_node_execute_errored else ""
+                message = create_error_message(kwargs, node_, step)
+                logger.exception(message)
                 raise
             finally:
-                if adapter.does_hook("post_node_execute", is_async=False):
-                    adapter.call_all_lifecycle_hooks_sync(
-                        "post_node_execute",
-                        run_id=run_id,
-                        node_=node_,
-                        kwargs=kwargs,
-                        success=success,
-                        error=error,
-                        result=result,
-                        task_id=task_id,
-                    )
+                if not pre_node_execute_errored and adapter.does_hook(
```

Comment:
Hmm we should really have a context manager that does this... But this works for now.